nessie/src/main/java/org/apache/iceberg/nessie/UpdateableReference.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.nessie;
+
+import com.dremio.nessie.api.TreeApi;
+import com.dremio.nessie.error.NessieNotFoundException;
+import com.dremio.nessie.model.Branch;
+import com.dremio.nessie.model.Hash;
+import com.dremio.nessie.model.Reference;
+import org.apache.iceberg.exceptions.NotFoundException;
+
+class UpdateableReference {
+
+  private Reference reference;
+  private final TreeApi client;
+
+  UpdateableReference(Reference reference, TreeApi client) {
+    this.reference = reference;
+    this.client = client;
+  }
+
+  public boolean refresh() {
+    if (reference instanceof Hash) {
+      return false;
+    }
+    Reference oldReference = reference;
+    try {
+      reference = client.getReferenceByName(reference.getName());
+    } catch (NessieNotFoundException e) {
+      throw new NotFoundException(e, "Failure refreshing data, table no longer exists.");
```

Comment:
Table no longer exists? Or the ref no longer exists?

Also, `NotFoundException` is for files that don't exist. Tables should use `NoSuchTableException`

Reply:
fixed the comment and simplified the exception handling